### PR TITLE
trigger error when config files are not readable

### PR DESCRIPTION
**Pull request by JohnRDOrazio**
fixes issue #9549

I just tested on my own installation and it is working correctly, when any config file has the wrong ownership I am now seeing in the logs:
```bash
[15-Jul-2024 18:03:36 UTC] PHP Warning:  /usr/share/psa-roundcube/config/config.inc.php is not readable. Please check ownership of file in /usr/share/psa-roundcube/program/lib/Roundcube/rcube_config.php on line 312
```

### CODE REVIEW

**Comment by JohnRDOrazio** [comment]
Perhaps even better would be to indicate the exact process user name and group that are needed for ownership:
```php
                    $processUser = posix_getpwuid(posix_geteuid());
                    $groupid   = posix_getegid();
                    $groupinfo = posix_getgrgid($groupid);
                    trigger_error($fpath . ' is not readable. Please check ownership of file, it should be owned by user ' . $processUser['name'] . ' or at least by group ' . $groupinfo['name'], \E_USER_WARNING);
```
I can make another commit if reviewers think that it would be a correct enhancement.

File: program/lib/Roundcube/rcube_config.php [hunk 1]
```diff
@@ -299,20 +299,25 @@ public function load_from_file($file)
...
-                if (isset($config) && is_array($config)) {
-                    $this->merge($config);
-                    $success = true;
-                }
-                // deprecated name of config variable
-                if (isset($rcmail_config) && is_array($rcmail_config)) {
-                    $this->merge($rcmail_config);
-                    $success = true;
+            if ($fpath && is_file($fpath)) {
+                if(false === is_readable($fpath)) {
```
Comment (pabzm) [inline program/lib/Roundcube/rcube_config.php:303]: Please insert a space before the first bracket here.

File: program/lib/Roundcube/rcube_config.php [hunk 2]
```diff
@@ -299,20 +299,25 @@ public function load_from_file($file)
...
+                    trigger_error($fpath . ' is not readable. Please check ownership of file', \E_USER_WARNING);
```
Comment (pabzm) [inline program/lib/Roundcube/rcube_config.php:304]: The problem might be due to other access restrictions than ownership, too. Maybe the message should be a little more general (or use just the first sentence, it has already all the relevant information).

**Review by pabzm — COMMENTED** [review]
Besides the two small nitpicks this looks good! Thank you for the contribution!

I wouldn't add more code to check for the owner and group names. If we do that it would have to work on different platforms and that's going to take more work, which is not really worth the effort, I think.

**Review by pabzm — CHANGES_REQUESTED** [review]
(placeholder to make the review marked as having requested changes, github doesn't allow me to change that after posting...)

File: program/lib/Roundcube/rcube_config.php (hunk 1 above)
Comment (JohnRDOrazio) [reply program/lib/Roundcube/rcube_config.php]: added missing space

File: program/lib/Roundcube/rcube_config.php (hunk 2 above)
Comment (JohnRDOrazio) [reply program/lib/Roundcube/rcube_config.php]: cut the error message to the first sentence

**Comment by JohnRDOrazio** [comment]
Does the second commit satisfy the requirements? Let me know if there are any other useful changes I can commit.

**Comment by JohnRDOrazio** [comment]
Should I fix the remaining coding style complaints? I see there is a rule for non-yoda-style. I had been encouraged by a developer on stackoverflow to start implementing yoda-style comparisons to make for better code debugging, but if yoda-style is considered less readable I can fix it.
I didn't touch the include but I see it's also complaining about the include using parentheses. I have started implementing `phpcs` in a few projects of my own, using the base `PSR12` ruleset which doesn't care about parentheses for includes. I can go ahead and remove the parentheses if that's considered more readable.

**Comment by alecpl** [comment]
Yes please. The rest looks good and it will be merged after the project's code style is used.

**Comment by JohnRDOrazio** [comment]
Code style is now passing, I'm now seeing that in the `origin/master` branch `require` is used rather than `include`, I don't believe I had changed anything in that area of the code but if you prefer that it match `master` I can set it to `require`.

**Comment by JohnRDOrazio** [comment]
I'm seeing that that change took place in commit https://github.com/roundcube/roundcubemail/commit/1e360999b261002ccdabb353bc951025179fb6e3, which goes back to April 11th. I guess my fork wasn't quite up to date...

**Comment by JohnRDOrazio** [comment]
If I try setting `include` to `require` my VSCode is telling me:
```
File is being conditionally included; use "include" instead
(PEAR.Files.IncludingFile.UseInclude)
```
Perhaps we can leave `include`?

**Comment by alecpl** [comment]
I prefer "require" here.

**Comment by JohnRDOrazio** [comment]
> I prefer "require" here.

Ok done

**Pull request merged**